src/adapter/src/catalog/apply.rs

```diff
@@ -1894,42 +1884,128 @@ fn sort_updates_inner(updates: Vec<StateUpdate>) -> Vec<StateUpdate> {
         }
     }
 
-    /// Sort item updates by [`CatalogItemId`].
+    /// Sort item updates by dependency.
```

Comment:
Mind adding some detail here about the sort order? Specifically how we sort by item type and then id, and historical context as to why we do this?

Reply:
Done